cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobManager.java

```diff
@@ -69,6 +69,16 @@ public interface RuntimeJobManager extends Closeable {
    */
   void kill(ProgramRunInfo programRunInfo) throws Exception;
 
+  /**
+   * Forcefully kills a running job. If the job is already in terminal status, then this method should be a no-op. If
+   * the job does not exist, this method should be a no-op.
+   *
+   * @param programRunInfo program run info
+   * @param runtimeJobStatus runtime Job status
+   * @throws Exception thrown if any exception while killing the job
+   */
+  void kill(ProgramRunInfo programRunInfo, RuntimeJobStatus  runtimeJobStatus) throws Exception;
```

Comment:
should pass in the RuntimeJobDetail, which has programRunInfo and status already.